### DISCUSSION TITLE
Does the community plan on adding more components?

### DISCUSSION THREAD

**Discussion by catpreneur** [system-ui/theme-ui#2265] [General]
Hi,

Just discovered this library not too long ago and I love it. I love using styled-components and styled-system and having a "headless" library of UI components built with sc/ss is superb!

My only question is: do the authors plan on adding more components? Here are some common components that many libraries have that theme-ui doesn't:

- Popover
- Dialog/Modal
- File uploader

Looking at the v1 roadmap https://github.com/system-ui/theme-ui/issues/832 doesn't look like more components are on the way anytime soon.

**Comment by hasparus** [comment]
I'm not sure if it makes sense to compete with excellent styleless component libraries.

I'm ususually combining Theme UI with [Radix](https://www.radix-ui.com/), [React Aria](https://react-spectrum.adobe.com/react-aria/getting-started.html) or [Reakit](https://reakit.io/docs/styling/).

@lachlanjc What do you think?

💡 Idea: Could we potentially replace `@theme-ui/components` with a "pattern library" website similar to Tailwind UI showing a complete implementation of a design system using Theme UI for styling and a headless accessible components library for the foundation?

**Comment by catpreneur** [comment]
I thought `@theme-ui/components` were headless components. Am i mistaken?

I think the built-in component primitives are pretty good to begin with, I was only highlighting a few must-have components that were missing :)
